File: Assignment37_1.java
//1.Write a program which checks whether 15th bit is On or OFF. 

import java.util.*;

class Assignment37_1
{
    public static void main(String Args[])
    {
        Scanner sobj = new Scanner(System.in);
        boolean bRet = false;

        System.out.println("Enter the Number Here : ");
        int iValue = sobj.nextInt();

        BitOpr bobj = new BitOpr(iValue);
        bRet = bobj.ChkBit();
        if(bRet==true)
        {
            System.out.println("The 15th Bit is ON....");
        }
        else
        {
            System.out.println("The 15th Bit is OFF....");
        }
        sobj.close();

    }
}

class BitOpr
{
    public int iNo;
    public int iMask;
    public int Result;

    public BitOpr(int A)
    {
        iNo = A;
        iMask = 0X00004000;
        Result = 0;
    }

    boolean ChkBit()
    {
        Result = iNo & iMask;

        if(Result == iMask)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

File: Assignment37_1.ts
//1.Write a program which checks whether 15th bit is On or OFF. 

class BitOpration
{
    public iNo : number;
    public iMask : number;
    public iResult : number;

    constructor(iVal : number)
    {
        this.iNo = iVal;
        this.iMask = 0X00004000;
        this.iResult = 0;
    }

    ChkBit() : boolean
    {
        this.iResult = this.iNo & this.iMask;
        if(this.iResult == this.iMask)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}


var bRet : boolean = false;

var Obj = new BitOpration(16384);
bRet = Obj.ChkBit()
if(bRet == true)
{
    console.log("The 15th Bit is On...");
}
else
{
    console.log("The 15th Bit is OFF...");
}


var Obj1 = new BitOpration(10000);
bRet = Obj1.ChkBit()
if(bRet == true)
{
    console.log("The 15th Bit is On...");
}
else
{
    console.log("The 15th Bit is OFF...");
}


